plugins/woocommerce/src/Internal/Admin/Analytics.php

```diff
@@ -285,4 +312,40 @@ public function run_clear_cache_tool() {
 
 		return __( 'Analytics cache cleared.', 'woocommerce' );
 	}
+
+	/**
+	 * "Fix" full refund data by re-importing all the refunded historical data.
+	 */
+	public function run_full_refund_fix_data_tool() {
+		global $wpdb;
+
+		Cache::invalidate();
+
+		// Get every order ID where:
+		// 1. the total sales is less than 0, and
+		// 2. is not refunded shipping fee only, and
+		// 3. is not refunded tax fee only.
+		$refunded_orders = $wpdb->get_results(
+			"SELECT order_stats.order_id
+			FROM {$wpdb->prefix}wc_order_stats AS order_stats
+			WHERE order_stats.total_sales < 0 # Refunded orders
+				AND order_stats.total_sales != order_stats.shipping_total # Exclude refunded orders that only include a shipping refund
+				AND order_stats.total_sales != order_stats.tax_total # Exclude refunded orders that only include a tax refund"
+		);
+
+		delete_option( 'woocommerce_analytics_uses_old_full_refund_data' );
```

Comment:
Should we `LIMIT` and paginate the orders, I am wondering on larger stores what would be size of the array?